crates/storage/provider/src/writer/mod.rs

```diff
@@ -322,7 +347,10 @@ impl<'a, 'b, DB: Database> StorageWriter<'a, 'b, DB> {
     }
 }
 
-impl<'a, 'b, DB: Database> StateWriter for StorageWriter<'a, 'b, DB> {
+impl<'a, 'b, TX> StateWriter for StorageWriter<'a, 'b, TX>
+where
+    TX: DbTxMut + DbTx,
+{
```

Comment:
unnecessary to combine writing receipts to static files and to db in one transaction, defeats the purpose a bit of having the separation between rw and ro tx in the first place for `StorageWriter`. would be better if this was separated into two functions called sequentially wherever `write_to_stroage` is called now. a little minus for dev ex but big plus for perf.

Reply:
I don't know if it's possible to split it up that way, or what the right API would be, because whether or not static files are written to depends on the prune config (ie in `append_receipts_from_blocks`). Open to ideas though